Posts/2021/12/jon-preferences.md

```diff
@@ -0,0 +1,258 @@
+---
+post_title: On Preferences and Scopes
+username: j@mobulaconsulting.com
+Categories: PowerShell
+tags: PowerShell, Preference Variables, Verbose, Progress, ErrorAction
+Summary: How to have more control of preferences in functions and the role of modules on inheritance. 
+---
+
+# Progress in PowerShell: a tale of Verbosity and other preferences with lessons in Scopes and Proxies thrown in
+
+It started, as these things often do, with someone complaining. In PowerShell Version 7.2 the output of `Invoke-WebRequest -Verbose` and `Invoke-RestMethod -Verbose` look like this:
+
+```powershell-console
+VERBOSE: GET with 0-byte payload
+```
+
+In all the earlier versions they look like the version below , which is more helpful when you're trying to debug code that builds URIs:
+
+```powershell-console
+VERBOSE: GET https://google.com/ with 0-byte payload
+```
+
+A *proxy function* will fix that. If two commands have the same name an alias beats a function,

```

Comment:
A *proxy function* can fix that. If two commands have the same name an alias beats a function,